### ISSUE TITLE
Fallback on Long Polling with .Net Core 2.1

### ISSUE THREAD

**Issue by z-tc** [aspnet/SignalR#3019]
I am using aspnet/SignalR 1.0.3 with .Net Core 2.1 (Microsoft.AspNetCore.SignalR 1.0.3 NuGet version). On connection, i am getting:
Error during WebSocket handshake: Unexpected response code: 500,
after that fallback to SSE also unsuccessful:
Error: Failed to start the transport 'ServerSentEvents': Error: Error occurred, 
and at last, i am connected successfully with Long Polling:
Client code:
public createConnection() {
        const options: any = {
            transport: 0,
            accessTokenFactory: () => 'my token'
        };
        this._hubConnection = new HubConnectionBuilder()
            .withUrl('http://localhost:5000/chathub', options )
            .build();
   }
When I set transport to 4 (long polling directly) - no errors.
Question is why first two (best) transports does not work? Might it be that that latest client package version does not support .Net Core 2.1? Any idea how to avoid this behavior currently?

Thanks!

**Comment by anurse** [comment]
Can you provide the environment details requested in the New Issue template? Also, collecting client-side logs as described in the [Diagnostics Guide](https://github.com/aspnet/SignalR/wiki/Diagnostics-Guide) will allow us to diagnose why those two transports are being skipped.

* The Server you are using (Kestrel/HttpSysServer/IIS/IIS Express/Azure Web App/etc.): 
* The Operating System on the Server (Windows/Linux/macOS):
* The Operating System on the Client (Windows/Linux/macOS):
* The Browser on the client, if using the JavaScript client (IE/Chrome/Edge/Firefox/etc.):

**Comment by z-tc** [comment]
Ok, i will turn the issue in another direction :) I found my issue - I am using [Authorize] attribute - authorization is going through IdentityServer4 Bearer token. When i send request with .Net client, i am setting:
`_hubConnection = new HubConnectionBuilder()
				.WithUrl("http://localhost:5000/chathub", options => 
					options.Headers.Add("Authorization", "Bearer myBearer"))
				.Build();`
With that setup, i am connected to websocket wuhout any problems. But, with angular client, i just can set accessTokenFactory in the scope of options, which pass the negotiation, but in request to the hub authorization header isn't set, just access_token in query, and it does not pass IdentityServer validation.

How  can I set authorization headers in client request for server hub methods usage?
